### PR TITLE
Add python wrappers for Quadrature

### PR DESCRIPTION
**Pull request by agrayver**
Part of #9015

### CODE REVIEW

File: contrib/python-bindings/include/quadrature_wrapper.h [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    /**
+     * Destructor.
+     */
+    ~QuadratureWrapper();
+
+    /**
+     * Create QGauss.
+     */
+    void
+    create_gauss(const unsigned int n);
```
Comment (masterleinad) [inline contrib/python-bindings/include/quadrature_wrapper.h]: Is there any reason not to templatize this to allow any kind of Quadrature formula? In the end, you only care for the quadrature points and weights anyway, don't you? You could still assert inside that the template type is one that is safe to be used, i.e. that doesn't have extra member variables.

Comment (agrayver) [reply contrib/python-bindings/include/quadrature_wrapper.h]: Not sure I understand. Are you suggesting to pass the quadrature type as a template argument? How to expose this to the user then? Also, many quadratures are constructed with additional parameters, so there is no universal constructor...

**Comment by agrayver** [comment]
@masterleinad thanks, I addressed your points.

File: contrib/python-bindings/include/quadrature_wrapper.h (hunk 1 above)
Comment (masterleinad) [reply contrib/python-bindings/include/quadrature_wrapper.h]: I was thinking about having something similar to what we do in [QuadratureSelector](https://www.dealii.org/developer/doxygen/deal.II/quadrature__selector_8cc_source.html#l00059) here.

Comment (agrayver) [reply contrib/python-bindings/include/quadrature_wrapper.h]: Personally, I don't see why this is better (there are many quadratures with ad-hoc constructor parameters). However, if you feel strong about that, I can do it.

**Comment by agrayver** [comment]
Can we move forward with this?

File: contrib/python-bindings/include/quadrature_wrapper.h (hunk 1 above)
Comment (masterleinad) [reply contrib/python-bindings/include/quadrature_wrapper.h]: No, I don't feel strongly about it. Let's move forward. We can always reevaluate before the next release.

**Pull request merged**